components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/EndpointConfigManager.java

```diff
@@ -0,0 +1,157 @@
+                    prop.load(inputStream);
+                    log.debug(Constants.TenantConstants.CONFIG_FILE_NAME
+                            + " file loaded from authentication endpoint webapp");
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Input stream is null while loading authentication endpoint from webapp");
+                    }
+                }
+            }
+
+            appName = getPropertyValue(Constants.CONFIG_APP_NAME);
+            appPassword = getPropertyValue(Constants.CONFIG_APP_PASSWORD);
+            isLocalTransportEnabled = Boolean.parseBoolean(getPropertyValue(Constants
+                    .CONFIG_LOCAL_TRANSPORT_ENABLED));
+            serverOrigin = IdentityUtil.fillURLPlaceholders(getPropertyValue(Constants.CONFIG_SERVER_ORIGIN));
+
+        } catch (IOException e) {
+            log.error("Initialization failed : ", e);
+        }
+    }
+
+    /**
+     * Get application name
+     *
+     * @return Application path
+     */
+    public static String getAppName() {
+
+        return appName;
+    }
+
+    /**
+     * Get application password
+     *
+     * @return Application password
+     */
+    public static String getAppPassword() {
+
+        return appPassword;
+    }
+
+    /**
+     * Check if local transport is enabled
+     *
+     * @return True if local transport is enabled
+     */
+    public static boolean isIsLocalTransportEnabled() {
+
+        return isLocalTransportEnabled;
+    }
+
+    /**
+     * Get server origin
+     *
+     * @return ServerOrigin
+     */
+    public static String getServerOrigin() {
+
+        return serverOrigin;
+    }
+
+    /**
+     * Build the absolute path of a give file path
+     *
+     * @param path File path
+     * @return Absolute file path
+     * @throws java.io.IOException
+     */
+    public static String buildFilePath(String path) throws IOException {
+
+        if (StringUtils.isNotEmpty(path) && path.startsWith(Constants.TenantConstants.RELATIVE_PATH_START_CHAR)) {
+            // Relative file path is given
+            File currentDirectory = new File(new File(Constants.TenantConstants.RELATIVE_PATH_START_CHAR)
+                    .getAbsolutePath());
+            path = currentDirectory.getCanonicalPath() + File.separator + path;
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("File path for KeyStore/TrustStore : " + path);
+        }
+        return path;
+    }
+
+    /**
+     * Get property value by key
+     *
+     * @param key Property key
+     * @return Property value
+     */
+    private static String getPropertyValue(String key) {
+        if ((Constants.SERVICES_URL.equals(key)) && !prop.containsKey(Constants.SERVICES_URL)) {
+            String serviceUrl = IdentityUtil.getServicePath();
+            return IdentityUtil.getServerURL(serviceUrl, true, true);
+        }
+        return prop.getProperty(key);
+    }
+}
```

Comment:
Why not we make this class a singleton rather than having public static methods configured

Reply:
In addition, with this class in place reading configs from https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/TenantDataManager.java#L74 should be deprecated and even TenantManager should refer to configs read from here.
At the moment let's create a git issue to track this improvement and proceed
Please refer the git issue here

Reply:
Looks like all initializers reads similarly.
Therefore, let's proceed with this model. But let's also create the git issue for refactoring later

Reply:
created [1] to track the suggested improvement

[1] https://github.com/wso2/product-is/issues/5265